### PR TITLE
Improve file edit screen

### PR DESCRIPTION
**Pull request by makeevrserg**
**Background**

File editor currently contains a lot of logic, which makes code reading harder. It can be improved by splitting it into separate decompose screen for each task: Loading, Editing, Uploading

**Changes**

- Split editor into multiple decompose screens

**Test plan**

- Open new editor. See new beautiful new loading screen with progress.
- After it loaded try edit your file
- After finish try save file or save file as new file.
- See new file immediately appeared in listing with updated size

### CODE REVIEW

File: components/filemngr/editor/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/editor/api/FileDownloadDecomposeComponent.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                    title = stringResource(FME.string.fme_status_downloading)
+                )
+            }
+
+            DownloadViewModel.State.TooLarge -> {
+                Box(Modifier.fillMaxSize().background(Color.Cyan))
+            }
+
+            DownloadViewModel.State.Unsupported -> {
+                Box(Modifier.fillMaxSize().background(Color.Yellow))
```
Comment (makeevrserg) [inline components/filemngr/editor/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/editor/api/FileDownloadDecomposeComponent.kt:82]: Shared errors will be added in next PRs.

File: components/filemngr/editor/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/editor/api/FileManagerEditorDecomposeComponentImpl.kt [hunk 2]
```diff
@@ -1,66 +1,102 @@
...
-        val editorViewModel = viewModelWithFactory(path.toString()) {
-            editorViewModelFactory.invoke(path)
+    @Assisted private val onFileChanged: (ListingItem) -> Unit,
+    fileDownloadDecomposeComponentFactory: FileDownloadDecomposeComponent.Factory,
+    private val editorDecomposeComponentFactory: EditorDecomposeComponent.Factory,
+    private val storageProvider: FlipperStorageProvider,
+    private val uploadFileDecomposeComponentFactory: UploadFileDecomposeComponent.Factory
+) : FileManagerEditorDecomposeComponent<FileManagerEditorConfiguration>(),
+    ComponentContext by componentContext {
+    private val editorFileKeeper = instanceKeeper.getOrCreate {
```
Comment (makeevrserg) [inline components/filemngr/editor/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/editor/api/FileManagerEditorDecomposeComponentImpl.kt:36]: The file should survive configuration change with other ViewModels. This file is used to keep changes we make inside file editor. The same file is uploaded on flipper after 'Save as' or 'Save' is clicked

File: components/filemngr/main/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/main/impl/api/FileManagerDecomposeComponentImpl.kt [hunk 3]
```diff
@@ -71,6 +72,12 @@ class FileManagerDecomposeComponentImpl @AssistedInject constructor(
                 componentContext = componentContext,
                 path = config.path,
                 onBack = { navigation.popOr(onBack::invoke) },
+                onFileChanged = { item ->
+                    val component = stack.findComponentByConfig(
+                        configClazz = FileManagerNavigationConfig.FileTree::class
+                    ) as? FilesDecomposeComponent
+                    component?.onFileChanged(item)
```
Comment (makeevrserg) [inline components/filemngr/main/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/main/impl/api/FileManagerDecomposeComponentImpl.kt:79]: Little doubtful, but any other good(and easy) way to notify stack component?

Comment (makeevrserg) [reply components/filemngr/main/impl/src/commonMain/kotlin/com/flipperdevices/filemanager/main/impl/api/FileManagerDecomposeComponentImpl.kt:79]: We have `Deeplinks`, but not events. Events may be recreated like somewhat [like this](https://gist.github.com/makeevrserg/0df667112292933d154e6bf748eccebc)
[Source 1](https://github.com/moffpage/cocktails/blob/material3-migration/core/src/commonMain/kotlin/kz/grandera/vlifetesttaskapp/core/extensions/ValueToStateFlow.kt)
[Source 2 from decompose chat](https://t.me/arkivanov_kt/22893)

**Review by LionZXY — APPROVED** [review]

**Pull request merged**